### PR TITLE
Add ERC-20 token contract and its tests

### PR DESCRIPTION
**Pull request by LOUDINISouad**

### CODE REVIEW

**Comment by achab** [comment]
Hi @LOUDINISouad, can you give a better name to the pull request? Like "Add ERC20 contract and its tests"

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: test
```
Comment (achab) [inline .github/workflows/test.yml:1]: @LOUDINISouad please remove this file

File: node_modules/.package-lock.json
Comment (achab) [inline node_modules/.package-lock.json:1]: this file should be removed. you should add `node_modules` to the file `.gitignore`

File: remappings.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+ds-test/=lib/forge-std/lib/ds-test/src/
```
Comment (achab) [inline remappings.txt:1]: why adding this file? is it part of Foundry framework?

**Comment by achab** [comment]
I've seen that you removed files about `Counter`. There's also a file in `scripts` about this contract. And you can also add the equivalent file for `MyToken` in `scripts` folder.

**Comment by achab** [comment]
Can you add a command in `README.md` to explain how to run tests with Foundry?

File: remappings.txt (hunk 2 above)
Comment (LOUDINISouad) [reply remappings.txt:1]: forge remappings > remappings.txt . I used this command after installing OpenZeppelin .

File: .github/workflows/test.yml (hunk 1 above)
Comment (achab) [reply .github/workflows/test.yml:1]: `git rm --cached .github/workflows/test.yml`

Comment (achab) [reply .github/workflows/test.yml:1]: we will add it in an upcoming PR. it's important to work on one topic per PR

File: package.json
Comment (achab) [inline package.json:1]: I don't think this file is needed, since it's not an npm project.

File: src/MyToken.sol [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+contract MyToken is ERC20 {
+    constructor() ERC20("MyToken", "LDN") {}
```
Comment (achab) [inline src/MyToken.sol:8]: the famous `LDN` token :)

File: test/MyToken.t.sol [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    }
+
+    function testInitialSupply() public {
+        assertEq(myToken.totalSupply(), 0, "Initial supply should be zero");
+    }
+
+    function testTokenNameAndSymbol() public {
+        assertEq(myToken.name(), "MyToken", "Token name is incorrect");
+        assertEq(myToken.symbol(), "LDN", "Token symbol is incorrect");
+    }
```
Comment (achab) [inline test/MyToken.t.sol:22]: could you also add a test case that shows that tranferring the token works well?

File: node_modules/.package-lock.json
Comment (achab) [inline node_modules/.package-lock.json:1]: this file should be removed

File: package-lock.json
Comment (achab) [inline package-lock.json:1]: this file should be removed

File: script/MyToken.s.sol
Comment (achab) [inline script/MyToken.s.sol:1]: the content of this file should be changed

**Pull request closed without merging**